osu.Game/Online/API/Requests/SearchBeatmapSetsRequest.cs

```diff
@@ -62,4 +73,38 @@ public enum BeatmapSearchCategory
         [Description("My Maps")]
         Mine,
     }
+
+    public enum BeatmapSearchGenre
+    {
+        Any,
+        Unspecified,
+
+        [Description("Video Game")]
+        Game,
+        Anime,
+        Rock,
+        Pop,
+        Other,
+        Novelty,
+
+        [Description("Hip Hop")]
+        Hiphop = 9,
+        Electronic
+    }
+
+    public enum BeatmapSearchLanguage
+    {
+        Any,
+        English = 2,
+        Chilnese = 4,
+        French = 7,
+        German,
+        Italian = 11,
+        Japanese = 3,
+        Korean = 6,
+        Spanish = 10,
+        Swedish = 9,
+        Instrumantal = 5,
+        Other = 1
+    }
```

Comment:
This looks super messy and in general prone to error (I had to stop and think why `German` had no value - it's because by accident it's after `French` which have both adjacent `language_id` and `display_order` web-side).

I'd rather [follow what web does entirely](https://github.com/ppy/osu-web/blob/ef8964813fa1d30e97cde9f0e7e117649b5cd9f4/database/seeds/ModelSeeders/MiscSeeder.php#L51-L64) by ordering _by value_ in the enum, and then adding some mechanism for display order (probably best to mirror directly and implement an `[Order(int)]` attribute).